### DISCUSSION TITLE
Debug renderer: more options for drawing sensors?

### DISCUSSION THREAD

**Discussion by eliasdaler** [jrouwe/JoltPhysics#1039] [Ideas]
Currently, sensors are always drawn in yellow and in wire mode:
![image](https://github.com/jrouwe/JoltPhysics/assets/1285136/277540e5-3148-4651-a772-be145ba44365)
It would be nice to have the ability to:
* Draw sensors in solid mode
* Have sensors be colored differently by instance

It would look like this: 
![image](https://github.com/jrouwe/JoltPhysics/assets/1285136/903bf6d6-bd93-492d-8402-9147ae1a93a8)

**Comment by jrouwe** [comment]
Hello,

I think it's a bit much for me to try to cater to everyone's wishes w.r.t. how the debug rendering should look. If you set DrawSettings::mDrawShape = false you can easily render the bodies however you want, e.g.:

```
BodyIDVector body_ids;
mPhysicsSystem->GetBodies(body_ids);
for (BodyID id : body_ids)
{
	BodyLockRead lock(mPhysicsSystem->GetBodyLockInterface(), id);
	if (lock.Succeeded())
	{
		const Body &body = lock.GetBody();
		if (body.IsSensor())
			body.GetShape()->Draw(DebugRenderer::sInstance, body.GetCenterOfMassTransform(), Vec3::sReplicate(1.0f), Color::sRed /* Pick a color you like */, false, false);
		else
			// Draw the non-sensors in the way you like...
	}
}
```

**Reply by eliasdaler** [reply]
Checked out `BodyManager::Draw` more closely and I see that it doesn't do that much magical stuff or call into any internal APIs after all. 

I guess I'll just implement my own, right. :)

Maybe it's worth mentioning somewhere in the docs that your drawing almost from scratch, because so far only this is mentioned:

> Finally call for example [PhysicsSystem::DrawBodies](https://jrouwe.github.io/JoltPhysics/class_physics_system.html#a8d4eda1c25d2c2e57a6964167870fe93) or [PhysicsSystem::DrawConstraints](https://jrouwe.github.io/JoltPhysics/class_physics_system.html#a4fa500f710dcc423062fd1514fe07780) to draw the state of the simulation.

... and this made me think that `DrawBodies` and `DrawConstraints` could access some internal state that you couldn't pull yourself. :)
